sdks/cpp/common/include/Device.h

```diff
@@ -138,7 +134,7 @@ class Device {
      */
     inline DetailLevel_e detail_level() const { return detail_level_; }
 
-    inline std::string getDefaultScope() const { return default_scope_.toString(); }
+    inline const std::string& getDefaultScope() const { return default_scope_.toString(); }
```

Comment:
does this return a reference to a deleted object? IDK but it depends on the implementation of default_scope_. please check